aten/src/ATen/templates/Tensor.h

```diff
@@ -129,6 +133,10 @@ struct Tensor : public detail::TensorBase {
   //example
   //Tensor * add(Tensor & b);
   ${tensor_method_declarations}
+
+  inline Tensor expand_as(const Tensor & t) const {
```

Comment:
Any reason these aren't native functions?  They won't end up in Declarations.yaml if you define them here.

Reply:
No reason, I think; I just took the path of least resistance for defining these.

A counter-question: how do we decide whether or not a method goes in `Tensor.h` or is a native function? There are a number of methods in `Tensor.h` and I don't really understand the calculation here.

Reply:
I don't think we should be adding more to Tensor.h; I'd view what is there now as the bootstrap (some of it deals with TH stuff that we don't want to deal with in native functions).

Reply:
The stuff in Tensor.h predates native functions, which is why they are not listed. New stuff should be native functions.  Before we move _everything_ over, we also need to make the native function declaration more concise and make sure the non-dynamic paths (that just drop the method into Tensor.h, for instance) works. @gchanan let me know if you want to chat about how to build out the native function interface.